.github/workflows/dusk_ci.yml

```diff
@@ -1,72 +1,29 @@
-on: [pull_request]
+on:
+  pull_request:
+  push:
+    branches:
+      - master
 
 name: Continuous integration
 
 jobs:
-  analyze:
-    name: Dusk Analyzer
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-      - uses: actions-rs/cargo@v1
-        with:
-          command: install
-          args: --git https://github.com/dusk-network/cargo-dusk-analyzer
-      - uses: actions-rs/cargo@v1
-        with:
-          command: dusk-analyzer
-
-  check:
-    name: Check
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
+  code_analysis:
+    name: Code Analysis
+    uses: dusk-network/.github/.github/workflows/code-analysis.yml@main
+    with:
+      clippy_default: false
+      clippy_args: --release -- -D warnings
 
-  test_nightly:
-    name: Nightly tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --release
+  dusk_analyzer:
+    name: Dusk Analyzer
+    uses: dusk-network/.github/.github/workflows/dusk-analysis.yml@main
 
-  test_nightly_rkyv:
-    name: Nightly tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --release --features rkyv-impl
+  test_std:
+    name: tests std
+    uses: dusk-network/.github/.github/workflows/run-tests.yml@main
 
-  fmt:
-    name: Rustfmt
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-      - run: rustup component add rustfmt
-      - uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+  test_rkyv:
+    name: tests rkyv
+    uses: dusk-network/.github/.github/workflows/run-tests.yml@main
+    with:
+      test_flags: --features rkyv-impl
```

Comment:
TBH, I find having special tests for `rkyv` a little strange. I think I understand why, but could it be possible to do this differently?

Reply:
@ureeves the `run-tests` job, by default, runs `cargo test --release` if no test flags are passed. 

If you want to also test a specific feature, you need to pass it like I did here.

You could get rid of either the `test_std` job or `rkyv-impl` if you're fine with running the test and not using it as a sort of "integration test" to run both with or without the feature enabled.

Reply:
Ah understood. The workflow is essentially a frontend to `cargo`, and we can pass flags as we please.

Reply:
Yup! For more details on how you can use it, see the README here: https://github.com/dusk-network/.github/tree/main/.github/workflows#readme

This is the underlying workflow being called for `cargo test`: https://github.com/dusk-network/.github/blob/main/.github/workflows/run-tests.yml